components/script/dom/servoparser/mod.rs

```diff
@@ -226,20 +249,23 @@ impl ServoParser {
         // the parser remains unsuspended.
         loop {
             self.document().reflow_if_reflow_timer_expired();
-            if let Err(script) = self.tokenizer.borrow_mut().feed(&mut *self.pending_input.borrow_mut()) {
-                if script.prepare() {
-                    continue;
-                }
-            }
+            let script = match self.tokenizer.borrow_mut().feed(&mut *self.pending_input.borrow_mut()) {
+                Ok(()) => break,
+                Err(script) => script,
+            };
+
+            let script_nesting_level = self.script_nesting_level.get();
 
-            // Document parsing is blocked on an external resource.
-            if self.suspended.get() {
+            self.script_nesting_level.set(script_nesting_level + 1);
+            script.prepare();
+            self.script_nesting_level.set(script_nesting_level);
+
+            if self.document.get_pending_parsing_blocking_script().is_some() {
+                self.suspend();
```

Comment:
This doesn't seem to match https://html.spec.whatwg.org/multipage/syntax.html#parsing-main-incdata:script-nesting-level-4 .